service/src/Helper.cpp

```diff
@@ -49,6 +49,28 @@ namespace geopm
         return contents;
     }
 
+    std::string read_file(const int fd)
+    {
+        std::string contents;
+        struct stat file_stat;
+
+        int err = fstat(fd, &file_stat);
+        if (err < 0) {
+            throw Exception("Helper::" + std::string(__func__) + "(): ",
+                            errno ? errno : GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        contents.resize(file_stat.st_size);
```

Comment:
This should put the null char at end of array.  Seems like an error to index into contents[contents.size()] on line 69.